core/include/scipp/core/eigen.h

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+#include <Eigen/Core>
+
+#include "scipp/core/dtype.h"
+
+namespace scipp::core {
+
+template <> inline constexpr DType dtype<Eigen::Vector3d>{4000};
+template <> inline constexpr DType dtype<Eigen::Matrix3d>{4001};
+template <> inline constexpr DType dtype<span<const Eigen::Vector3d>>{4100};
```

Comment:
So there is no `span` for `Matrix3d`? Would that mean you can't slice a variable with `dtype<Matrix3d>`?

Reply:
This affects anything that uses `subspan` view, which is mostly `bin`, `rebin`, and `histogram`. Adding this later should not be a problem and we should probably do so as part of a systematic review of the dtypes supported by all our operations.